src/main/java/org/spongepowered/api/util/Color.java

```diff
@@ -85,33 +107,39 @@
      * @param hex The hexadecimal value of the color
      * @return The color object
      */
-    public static Color ofRgb(int hex) {
-        return ofRgb((hex >> 0x10) & MASK, (hex >> 0x8) & MASK, hex & MASK);
+    public static Color ofHex(int hex) {
+        checkArgument(0 <= hex && hex <= 0xFFFFFF, "hex value [%s] must be in range 0 to 0xFFFFFF (16777215)", hex);
```

Comment:
Not sure about this range check as its pretty common to have hexcodes with alpha values, better to just ignore the alpha than requiring it to be zero.

Reply:
I don't particularly have a preference on this, but I think it's probably best to throw an exception if alpha is defined to make it clear that it isn't supported. If we were to support alpha, we could either strip it away (which in my opinion masks the issue and could cause problems with converting to java colors) or keep it, which begs the question of what do we do with alpha values for `#getRgb`, `#equals`, and `#hashCode`?